src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java

```diff
@@ -225,4 +229,81 @@ private void setupMultiselectField(MaintenanceDocument document, MaintenanceDocu
         }
         
     }
+    
+    @Override
+    public void doRouteStatusChange(DocumentHeader header) {
+        LOG.debug("doRouteStatusChange: entering");
+        super.doRouteStatusChange(header);
+        VendorDetail vendorDetail = (VendorDetail) getBusinessObject();
+        WorkflowDocument workflowDoc = header.getWorkflowDocument();
+        
+        if (workflowDoc.isProcessed()) {
+            LOG.debug("doRouteStatusChange: workflow is processed");
+            if (recordPvenRouteStatusChangeForPaymentWorksIsEnabled()) {
+                performPaymentWorksApprovalProcessingForVendor(vendorDetail, this.getDocumentNumber(), this.getMaintenanceAction());
+            }
+        }
+        else if (workflowDoc.isDisapproved() || workflowDoc.isCanceled()) {
+            LOG.debug("doRouteStatusChange: disapproved or canceled");
+            if (recordPvenRouteStatusChangeForPaymentWorksIsEnabled()) {
+                performPaymentWorksDisapprovalCancelProcessingForVendor(vendorDetail, this.getDocumentNumber(), this.getMaintenanceAction());
+            }
+        }
+    }
+    
+    private void performPaymentWorksApprovalProcessingForVendor(VendorDetail vendorDetail, String kfsDocumentNumber, String maintenanceAction) {
+        if (isExistingPaymentWorksVendor(kfsDocumentNumber)) {
+            LOG.debug("performPaymentWorksApprovalProcessingForVendor: isExistingPaymentWorksVendor");
+            processExistingPaymentWorksVendorApproval(kfsDocumentNumber, vendorDetail);
+        }
+        else if (StringUtils.equals(KRADConstants.MAINTENANCE_NEW_ACTION, maintenanceAction) ||
+                 StringUtils.equals(KRADConstants.MAINTENANCE_NEWWITHEXISTING_ACTION, maintenanceAction)) {
+            LOG.debug("performPaymentWorksApprovalProcessingForVendor: new action resulting from AVf or hand keying");
+            processKfsVendorNewActionApproval(kfsDocumentNumber, vendorDetail);
+        }
+        else if (StringUtils.equals(KRADConstants.MAINTENANCE_EDIT_ACTION, maintenanceAction)) {
+            LOG.debug("performPaymentWorksApprovalProcessingForVendor: edit action");
+            processKfsVendorEditActionApproval(kfsDocumentNumber, vendorDetail);
+        }
+    }
+    
+    private boolean isExistingPaymentWorksVendor(String kfsDocumentNumber) {
+        return getPaymentWorksBatchUtilityService().foundExistingPaymentWorksVendorByKfsDocumentNumber(kfsDocumentNumber);
+    }
+    
+    private boolean recordPvenRouteStatusChangeForPaymentWorksIsEnabled() {
+        return (getPaymentWorksBatchUtilityService().retrievePaymentWorksParameterValue(PaymentWorksParameterConstants.PMW_INTEGRATION_IS_ACTIVE_IND).equalsIgnoreCase(KFSConstants.ParameterValues.YES) ? true : false);
```

Comment:
You're already getting a boolean result from this check, so adding a ternary operation of "? true : false" is redundant.

Reply:
You are correct. Removing the ternary operator.